vmm/src/config.rs

```diff
@@ -2157,12 +2192,97 @@ impl RestoreConfig {
             .map_err(Error::ParseRestore)?
             .unwrap_or(Toggle(false))
             .0;
+        let net_fds = parser
+            .convert::<Tuple<String, Vec<u64>>>("net_fds")
+            .map_err(Error::ParseRestore)?
+            .map(|v| {
+                v.0.iter()
+                    .map(|(id, fds)| RestoredNetConfig {
+                        id: id.clone(),
+                        num_fds: fds.len(),
+                        fds: Some(fds.iter().map(|e| *e as i32).collect()),
+                    })
+                    .collect()
+            });
 
         Ok(RestoreConfig {
             source_url,
             prefault,
+            net_fds,
         })
     }
+
+    pub fn validate(&self, vm_config: &VmConfig) -> ValidationResult<()> {
+        // Check if multiple same ids are passed
+        // verify if passed ids are valid
+        let vm_net_ids = match &vm_config.net {
+            // It's okay to unwrap id because it is guaranteed to have a net id in the snapshot config.
+            Some(net_configs) => net_configs
+                .iter()
+                .map(|v| v.id.as_ref().unwrap().clone())
+                .collect(),
+            None => Vec::new(),
+        };
+        if let Some(nets) = &self.net_fds {
+            let mut net_ids_set = BTreeSet::new();
+            for net in nets.iter() {
+                if !vm_net_ids.contains(&net.id) {
+                    return Err(ValidationError::InvalidIdentifier(net.id.clone()));
+                }
+                if !net_ids_set.insert(net.id.clone()) {
+                    return Err(ValidationError::IdentifierNotUnique(net.id.clone()));
+                }
+            }
+        }
+        // Iterate through the net_ids and check if all required ids are passed without any unexpected ids
```

Comment:
I think this comment might need updating.

Reply:
Updated the comment.